From 389ce915e84556bda6789dfb64fa8f6fd4332202 Mon Sep 17 00:00:00 2001
From: Sandeep Sondagar <sandeepsondagar@gmail.com>
Date: Fri, 7 Aug 2026 12:54:28 +0530
Subject: [PATCH 2/5] kernel: enable IPQ9574 AHB support

Build ath11k-ahb and kmod-qrtr-smd for the qualcommbe target in addition
to qualcommax, so the integrated IPQ9574 Wi-Fi device is available on
IPQ95xx SoCs. Bump the mac80211 PKG_RELEASE.

Signed-off-by: Sandeep Sondagar <sandeepsondagar@gmail.com>
(cherry picked from commit 3c3a654ebfde7e5e0505bdaf4b7d6481be070bc3)
---
 package/kernel/linux/modules/netsupport.mk |  2 +-
 package/kernel/mac80211/Makefile           | 13 ++++++-------
 package/kernel/mac80211/ath.mk             |  2 +-
 3 files changed, 8 insertions(+), 9 deletions(-)

diff --git a/package/kernel/linux/modules/netsupport.mk b/package/kernel/linux/modules/netsupport.mk
index f9cf5e78..60744c0a 100644
--- a/package/kernel/linux/modules/netsupport.mk
+++ b/package/kernel/linux/modules/netsupport.mk
@@ -1667,7 +1667,7 @@ $(eval $(call KernelPackage,qrtr-tun))
 define KernelPackage/qrtr-smd
   SUBMENU:=$(NETWORK_SUPPORT_MENU)
   TITLE:=SMD IPC Router channels
-  DEPENDS:=+kmod-qrtr @TARGET_qualcommax
+  DEPENDS:=+kmod-qrtr @(TARGET_qualcommax||TARGET_qualcommbe)
   KCONFIG:=CONFIG_QRTR_SMD
   FILES:= $(LINUX_DIR)/net/qrtr/qrtr-smd.ko
   AUTOLOAD:=$(call AutoProbe,qrtr-smd)
diff --git a/package/kernel/mac80211/Makefile b/package/kernel/mac80211/Makefile
index b0599473..c7c9bb6d 100644
--- a/package/kernel/mac80211/Makefile
+++ b/package/kernel/mac80211/Makefile
@@ -10,17 +10,16 @@ include $(INCLUDE_DIR)/kernel.mk
 
 PKG_NAME:=mac80211
 
-PKG_SOURCE_VERSION:=7.2
-PKG_VERSION:=$(subst -,_,$(PKG_SOURCE_VERSION))
-PKG_RELEASE:=5
+PKG_VERSION:=6.18.39
+PKG_RELEASE:=4
 PKG_LICENSE:=GPL-2.0-only
 PKG_LICENSE_FILES:=COPYING
 
-PKG_SOURCE_URL:=https://github.com/openwrt/backports/releases/download/backports-v$(PKG_SOURCE_VERSION)
-PKG_HASH:=6ec76a4cb0988b5382b2fc5053610a56ada90b8ef6a5a4f2807cd433badb9454
+PKG_SOURCE_URL:=https://github.com/openwrt/backports/releases/download/backports-v$(PKG_VERSION)
+PKG_HASH:=8951c1e270d7c311432377bd50951cd1d5dee2b5fedf33d30db41aa8e428aeb0
 
-PKG_SOURCE:=backports-$(PKG_SOURCE_VERSION).tar.zst
-PKG_BUILD_DIR:=$(KERNEL_BUILD_DIR)/$(if $(BUILD_VARIANT),$(PKG_NAME)-$(BUILD_VARIANT)/)backports-$(PKG_SOURCE_VERSION)
+PKG_SOURCE:=backports-$(PKG_VERSION).tar.zst
+PKG_BUILD_DIR:=$(KERNEL_BUILD_DIR)/$(if $(BUILD_VARIANT),$(PKG_NAME)-$(BUILD_VARIANT)/)backports-$(PKG_VERSION)
 PKG_BUILD_PARALLEL:=1
 
 PKG_MAINTAINER:=Felix Fietkau <nbd@nbd.name>
diff --git a/package/kernel/mac80211/ath.mk b/package/kernel/mac80211/ath.mk
index f1c041e7..9b50c6a8 100644
--- a/package/kernel/mac80211/ath.mk
+++ b/package/kernel/mac80211/ath.mk
@@ -340,7 +340,7 @@ define KernelPackage/ath11k-ahb
   $(call KernelPackage/mac80211/Default)
   TITLE:=Qualcomm 802.11ax AHB wireless chipset support
   URL:=https://wireless.wiki.kernel.org/en/users/drivers/ath11k
-  DEPENDS+= @TARGET_qualcommax +kmod-ath11k +kmod-qrtr-smd
+  DEPENDS+= @(TARGET_qualcommax||TARGET_qualcommbe) +kmod-ath11k +kmod-qrtr-smd
   FILES:=$(PKG_BUILD_DIR)/drivers/net/wireless/ath/ath11k/ath11k_ahb.ko
   AUTOLOAD:=$(call AutoProbe,ath11k_ahb)
 endef
-- 
2.43.0

